### PR TITLE
Feat(impl)/225 use caching for discovery finder requests

### PR DESCRIPTION
**Pull request by dsmf**

### CODE REVIEW

**Comment by dsmf** [comment]
https://github.com/eclipse-tractusx/item-relationship-service/issues/225

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]
Some minor changes requested. Everything else looks good!

File: irs-api/src/test/java/org/eclipse/tractusx/irs/DiscoveryFinderClientTest.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+        // WHEN
+        final var actualResult = testee.findDiscoveryEndpoints(request);
+
+        // THEN
+        // real endpoint must be called
+        verify(restTemplateMock).postForObject("", request, DiscoveryResponse.class);
+
+        // and the response must be cached
+        {
```
Comment (ds-psosnowski) [inline irs-api/src/test/java/org/eclipse/tractusx/irs/DiscoveryFinderClientTest.java:87]: btw. what is the point of scoping with `{}`?

**Review by ds-psosnowski — DISMISSED** [review]

File: irs-api/src/test/java/org/eclipse/tractusx/irs/DiscoveryFinderClientTest.java (hunk 1 above)
Comment (dsmf) [reply irs-api/src/test/java/org/eclipse/tractusx/irs/DiscoveryFinderClientTest.java:87]: Technically none. But I use this in tests sometimes in order to group a block of asserts that belong together. This way it is sufficient for the reader to read only the comment above the {} block to understand what the test is about to assert. In real production code I'd rather use extract-to-method here, but in test I find it more helpful this way (unless the same asserts are used in other tests).

File: docs/src/docs/administration/configuration.adoc [hunk 2]
```diff
@@ -74,8 +74,15 @@ The hostname where Grafana will be made available.
 The EDC consumer controlplane endpoint URL for data management, including the protocol.
 If left empty, this defaults to the internal endpoint of the controlplane provided by the irs-edc-consumer Helm chart.
 
+==== <discoveryFinderClient.cacheTTL>
+
+When IRS calls the DTR discovery finder URL for BPNL's there is a cache mechanism between them, to improve performance.
```
Comment (ds-jhartmann) [inline docs/src/docs/administration/configuration.adoc:79]:
```suggestion
When IRS calls the Discovery Finder URL for BPNLs, the results are cached to improve performance.
```

File: docs/src/docs/administration/configuration.adoc [hunk 3]
```diff
@@ -74,8 +74,15 @@ The hostname where Grafana will be made available.
...
+This parameter defines how long the cache is maintained before it is cleared.
+Data is in ISO 8601.
+
 ==== <connectorEndpointService.cacheTTL>
-When IRS calls EDC Discovery Service to fetch endpoints for BPNL's there is a cache mechanism between them, to improve performance.
+
+When IRS calls EDC Discovery Service to fetch connector endpoints for BPNL's there is a cache mechanism between them, to improve performance.
```
Comment (ds-jhartmann) [inline docs/src/docs/administration/configuration.adoc:85]:
```suggestion
When IRS calls EDC Discovery Service to fetch connector endpoints for BPNLs, the results are cached to improve performance.
```

File: docs/src/docs/arc42/cross-cutting/under-the-hood.adoc [hunk 4]
```diff
@@ -97,3 +97,14 @@ Whenever a BPN is resolved via BPDM, the partner name is cached on IRS side, as
...
+=== Discovery Service
+
+The IRS uses the Discovery Finder in order to find the correct EDC Discovery URL for the type BPN.
+This URL is cached locally.
+
+When the EDC Discovery is requested to return the EDC connector endpoint URLs for a specific BPN, the results are cached as well.
+
+The time to live for both caches can be configured separately as described in the Administration Guide.
+
+Please find more information on Discovery Service see chapter "System scope and context".
```
Comment (ds-jhartmann) [inline docs/src/docs/arc42/cross-cutting/under-the-hood.adoc:110]:
```suggestion
Further information on Discovery Service can be found in the chapter "System scope and context".
```

**Review by ds-jhartmann — COMMENTED** [review]
Some minor grammatical suggestions

File: docs/src/docs/arc42/cross-cutting/under-the-hood.adoc (hunk 4 above)
Comment (dsmf) [reply docs/src/docs/arc42/cross-cutting/under-the-hood.adoc:110]: Hi @ds-jhartmann , 

Ah yes, the "see" was a mistake. Completely overlooked that when I rewrote the sentence. 

However the term "Please find more information" is a correct formal term AFAIK:  https://www.linguee.de/englisch-deutsch/uebersetzung/please+find+more+information.html

To keep things short I'll just use your suggestion.

**Review by ds-jhartmann — APPROVED** [review]

**Pull request merged**